src/lib.rs

```diff
@@ -509,12 +509,11 @@ impl<TypeState> ConnectionOptions<TypeState> {
     ///
     /// # Examples
     /// ```no_run
-    /// # fn main() -> std::io::Result<()> {
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
this allows most errors to be converted into the return type using the try `?` operator